File: examples/javascript/12-domain-policy-packs/domain-policy-packs.mjs
// Gavio Domain Policy Pack Catalog.

import { Gateway } from 'gavio'
import {
  listPolicyPacks,
  loadPolicyPack,
  piiGuardFromPolicyPack,
} from 'gavio/interceptors/pii'

const healthcare = loadPolicyPack('healthcare')
const india = loadPolicyPack('regional/india')
const hr = loadPolicyPack('hr').withOverrides({
  detectors: {
    employee_id: {
      action: 'flag',
      severity: 'critical',
      redactionStrategy: 'hash',
    },
  },
})

console.log('catalog :', listPolicyPacks().join(', '))
console.log('signed  :', healthcare.id, healthcare.verifySignature())
console.log('override:', hr.manifest().detectors.find((d) => d.name === 'employee_id'))

const gw = new Gateway({ devMode: true }).use(
  piiGuardFromPolicyPack([healthcare, india, hr], { logEntityTypes: false }),
)

const r = await gw.complete({
  messages: [
    {
      role: 'user',
      content:
        'Patient MRN-123456 has member MEM-AB12CD34. PAN ABCDE1234F and Aadhaar 1234 5678 9012 are present. Template EMP-000000 is allowed, but EMP-123456 is real.',
    },
  ],
})

console.log('reply    :', r.content)
console.log('PII found:', [...r.audit.piiEntityTypes].sort())

File: examples/python/12-domain-policy-packs/domain_policy_packs.py
"""Gavio Domain Policy Pack Catalog.

    pip install -r requirements.txt
    python domain_policy_packs.py
"""

import asyncio

from gavio import Gateway
from gavio.interceptors.pii import PiiGuard, PolicyPack, list_policy_packs


async def main() -> None:
    healthcare = PolicyPack.load("healthcare")
    india = PolicyPack.load("regional/india")
    hr = PolicyPack.load("hr").with_overrides(
        {
            "detectors": {
                "employee_id": {
                    "action": "flag",
                    "severity": "critical",
                    "redactionStrategy": "hash",
                }
            }
        }
    )

    print("catalog :", ", ".join(list_policy_packs()))
    print("signed  :", healthcare.id, healthcare.verify_signature())
    print("override:", next(d for d in hr.manifest()["detectors"] if d["name"] == "employee_id"))

    gw = (
        Gateway.builder()
        .dev_mode(True)
        .use(PiiGuard.from_policy_pack(healthcare, india, hr, log_entity_types=False))
        .build()
    )
    resp = await gw.complete(
        messages=[
            {
                "role": "user",
                "content": (
                    "Patient MRN-123456 has member MEM-AB12CD34. "
                    "PAN ABCDE1234F and Aadhaar 1234 5678 9012 are present. "
                    "Template EMP-000000 is allowed, but EMP-123456 is real."
                ),
            }
        ]
    )

    print("reply    :", resp.content)
    print("PII found:", sorted(resp.audit.pii_entity_types))


if __name__ == "__main__":
    asyncio.run(main())
